src/dxvk/rtx_render/rtx_types.cpp

```diff
@@ -128,39 +128,46 @@ namespace dxvk {
     }
   }
 
+  static bool wasCategorySetViaD3D(InstanceCategories category, uint32_t categoryFlags) {
+    return categoryFlags & (1 << static_cast<uint32_t>(category));
+  }
+
   void DrawCallState::setupCategoriesForTexture() {
     // TODO (REMIX-231): It would probably be much more efficient to use a map of texture hash to category flags, rather
     //                   than doing N lookups per texture hash for each category.
     const XXH64_hash_t& textureHash = materialData.getColorTexture().getImageHash();
 
-    setCategory(InstanceCategories::WorldUI, lookupHash(RtxOptions::worldSpaceUiTextures(), textureHash));
-    setCategory(InstanceCategories::WorldMatte, lookupHash(RtxOptions::worldSpaceUiBackgroundTextures(), textureHash));
+    const auto& d3dCategoryEnabled = RtxOptions::Get()->useUnusedRenderstates();
```

Comment:
generally prefer to just have this marked as a bool, to make it clearer exactly what is going on